libcxx/include/__config

```diff
@@ -325,6 +325,11 @@
 // clang-format on
 
 #  ifndef _LIBCPP_HARDENING_MODE
+
+#    ifndef _LIBCPP_HARDENING_MODE_DEFAULT
+#      error _LIBCPP_HARDENING_MODE_DEFAULT is not defined.
```

Comment:
Do we want to speculate in the comment as to why that might be the case? E.g.
```cpp
// Normally this macro comes from the `__config_site` header which is generated via CMake.
// If you use a custom build system, please make sure to define `_LIBCPP_HARDENING_MODE_DEFAULT`
// to one of the valid hardening modes.
```

Reply:
I would say something like `This definition should be set at configuration-time in the __config_site header, please ensure your installation of libc++ is not broken` -- or something like that. WDYT?

Reply:
Done.